### DISCUSSION TITLE
Failed to deploy milvus cluster offline in k8s cluster

### DISCUSSION THREAD

**Discussion by AliscaCL** [milvus-io/milvus#15918] [Q&A and General discussion]
Hello~ I am following the page https://milvus.io/docs/v2.0.0/install_offline-docker.md to install milvus cluster mode to deploy the milvus cluster. 
But the cluster not started successfully. 
The display of kubect get pods is shown as follows.

The still keep init pods will shown the logs as follows:

Error from server(BadRequest): container "opinion-release-pulsar-zookeeper-metadata" in pod "opinion-release-pulsar-zookeeper-metadata-tnshd" is waiting to start: PodInitializing.

**Comment by AliscaCL** [comment]
The pods might been shown as follows:
![微信图片_20220308090452](https://user-images.githubusercontent.com/96522752/157147044-101829fb-5e0f-4898-89f5-df1f3e4793f4.jpg)

**Reply by jennyli-z** [reply]
Could you use `kubectl describe pod ` command to get the events for the pod to see what happened?

**Reply by AliscaCL** [reply]
Which pod should I show for you？

**Reply by AliscaCL** [reply]
Hello, I have simulate a environment in Internet environment. But the backend of kubectl is minikube. 
The describe of opinion-release-pulsar-autorecovery-64665687b-fswq4 is

Name:         opinion-release-pulsar-autorecovery-64665687b-fswq4
Namespace:    default
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Tue, 08 Mar 2022 13:59:44 +0800
Labels:       app=pulsar
              cluster=opinion-release-pulsar
              component=autorecovery
              pod-template-hash=64665687b
              release=opinion-release
Annotations:  <none>
Status:       Pending
IP:           172.17.0.18
IPs:
  IP:           172.17.0.18
Controlled By:  ReplicaSet/opinion-release-pulsar-autorecovery-64665687b
Init Containers:
  wait-zookeeper-ready:
    Container ID:  docker://a7230af4708bee031e8669659805e401fbbf91cbd47f02cd7b377ecbb65a3982
    Image:         apachepulsar/pulsar:2.8.2
    Image ID:      docker-pullable://apachepulsar/pulsar@sha256:d538416d5afe03360e10d5beb44bdad33d7303d137fc66c264108426875f61c6
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
    Args:
      until bin/pulsar zookeeper-shell -server opinion-release-pulsar-zookeeper ls /admin/clusters | grep "^\[.*opinion-release-pulsar.*\]"; do
        sleep 3;
      done;
    State:          Running
      Started:      Tue, 08 Mar 2022 14:02:05 +0800
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-6xs5d (ro)
Containers:
  opinion-release-pulsar-autorecovery:
    Container ID:  
    Image:         apachepulsar/pulsar:2.8.2
    Image ID:      
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
    Args:
      bin/apply-config-from-env.py conf/bookkeeper.conf && bin/apply-config-from-env.py conf/bkenv.sh && exec bin/bookkeeper autorecovery
      
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:     1
      memory:  512Mi
    Environment Variables from:
      opinion-release-pulsar-autorecovery  ConfigMap  Optional: false
    Environment:                           <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-6xs5d (ro)
Conditions:
  Type              Status
  Initialized       False 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  kube-api-access-6xs5d:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age    From               Message
  ----    ------     ----   ----               -------
  Normal  Scheduled  3m47s  default-scheduler  Successfully assigned default/opinion-release-pulsar-autorecovery-64665687b-fswq4 to minikube
  Normal  Pulling    3m45s  kubelet            Pulling image "apachepulsar/pulsar:2.8.2"
  Normal  Pulled     86s    kubelet            Successfully pulled image "apachepulsar/pulsar:2.8.2" in 2m18.333363848s
  Normal  Created    86s    kubelet            Created container wait-zookeeper-ready
  Normal  Started    85s    kubelet            Started container wait-zookeeper-ready

**Reply by AliscaCL** [reply]
I check if I only run the milvus with minimum mode, still get the error logs:

Error from server (BadRequest): container "opinion-release-pulsar-bookkeeper" in pod "opinion-release-pulsar-bookkeeper-0" is waiting to start: PodInitializing

Meanwhile, the result of kubectl get pods is:

NAME                                  READY   STATUS     RESTARTS   AGE
opinion-release-etcd-0                1/1     Running    0          78m
opinion-release-etcd-1                1/1     Running    0          78m
opinion-release-etcd-2                1/1     Running    0          78m
opinion-release-minio-0               0/1     Pending    0          78m
opinion-release-minio-1               0/1     Pending    0          78m
opinion-release-minio-2               0/1     Pending    0          78m
opinion-release-minio-3               0/1     Pending    0          78m
opinion-release-pulsar-bookkeeper-0   0/1     Init:0/2   0          78m

**Reply by jennyli-z** [reply]
> `Error from server (BadRequest): container "opinion-release-pulsar-bookkeeper" in pod "opinion-release-pulsar-bookkeeper-0" is waiting to start: PodInitializing` 
The error message means  the pod is in the progress, need to wait for a while.
From the pod list status, the minio is also pending,  probably you need to check if pvc is created by using command `kubectl get pvc`  and check the pvc status is Bound or not?

**Reply by AliscaCL** [reply]
Hello, sorry for missing the message. 
Here is the info for pv and pvc, it seems nothing wrong.
(squad) alisca@LAPTOP-P7T2T6PL:~/PyProject/milvus-offline/images$ kubectl get pvc -n opinion
NAME                                                                            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-opinion-release-etcd-0                                                     Bound    pvc-0695dbfb-f974-46b8-9e8b-de1dca686193   10Gi       RWO            standard       4m39s
data-opinion-release-etcd-1                                                     Bound    pvc-54de8cd1-8e0a-49de-8d2b-d938c38eb5bb   10Gi       RWO            standard       4m39s
data-opinion-release-etcd-2                                                     Bound    pvc-83451c50-0279-42f1-b201-e5ddfcbe8ae1   10Gi       RWO            standard       4m39s
export-opinion-release-minio-0                                                  Bound    pvc-82df7123-0c09-49aa-a400-e96d0798a915   500Gi      RWO            standard       4m39s
export-opinion-release-minio-1                                                  Bound    pvc-b0405596-b013-439c-b3a8-ce6b84b26685   500Gi      RWO            standard       4m39s
export-opinion-release-minio-2                                                  Bound    pvc-253d2120-0a3a-41dd-af24-e3bce30c966e   500Gi      RWO            standard       4m39s
export-opinion-release-minio-3                                                  Bound    pvc-add12c94-246a-40f5-8192-a684698db0f7   500Gi      RWO            standard       4m38s
opinion-release-pulsar-bookkeeper-journal-opinion-release-pulsar-bookkeeper-0   Bound    pvc-e4edf03f-293b-4a2d-be66-1ca786e1e9d2   100Gi      RWO            standard       4m39s
opinion-release-pulsar-bookkeeper-ledgers-opinion-release-pulsar-bookkeeper-0   Bound    pvc-69f7a0cc-520c-473a-9f7e-2dbf11176fd8   200Gi      RWO            standard       4m39s
opinion-release-pulsar-zookeeper-data-opinion-release-pulsar-zookeeper-0        Bound    pvc-27873d28-aa30-4f99-a136-daa128101cc0   5Gi        RWO            standard       4m39s
(squad) alisca@LAPTOP-P7T2T6PL:~/PyProject/milvus-offline/images$ kubectl get pv -n opinion
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                                                                                   STORAGECLASS   REASON   AGE
pvc-0695dbfb-f974-46b8-9e8b-de1dca686193   10Gi       RWO            Delete           Bound    opinion/data-opinion-release-etcd-0                                                     standard                4m48s
pvc-253d2120-0a3a-41dd-af24-e3bce30c966e   500Gi      RWO            Delete           Bound    opinion/export-opinion-release-minio-2                                                  standard                4m47s
pvc-27873d28-aa30-4f99-a136-daa128101cc0   5Gi        RWO            Delete           Bound    opinion/opinion-release-pulsar-zookeeper-data-opinion-release-pulsar-zookeeper-0        standard                4m48s
pvc-54de8cd1-8e0a-49de-8d2b-d938c38eb5bb   10Gi       RWO            Delete           Bound    opinion/data-opinion-release-etcd-1                                                     standard                4m48s
pvc-69f7a0cc-520c-473a-9f7e-2dbf11176fd8   200Gi      RWO            Delete           Bound    opinion/opinion-release-pulsar-bookkeeper-ledgers-opinion-release-pulsar-bookkeeper-0   standard                4m48s
pvc-82df7123-0c09-49aa-a400-e96d0798a915   500Gi      RWO            Delete           Bound    opinion/export-opinion-release-minio-0                                                  standard                4m48s
pvc-83451c50-0279-42f1-b201-e5ddfcbe8ae1   10Gi       RWO            Delete           Bound    opinion/data-opinion-release-etcd-2                                                     standard                4m47s
pvc-add12c94-246a-40f5-8192-a684698db0f7   500Gi      RWO            Delete           Bound    opinion/export-opinion-release-minio-3                                                  standard                4m47s
pvc-b0405596-b013-439c-b3a8-ce6b84b26685   500Gi      RWO            Delete           Bound    opinion/export-opinion-release-minio-1                                                  standard                4m48s
pvc-e4edf03f-293b-4a2d-be66-1ca786e1e9d2   100Gi      RWO            Delete           Bound    opinion/opinion-release-pulsar-bookkeeper-journal-opinion-release-pulsar-bookkeeper-0   standard                4m48s

**Reply by jennyli-z** [reply]
1.  what's the status for the pod `opinion-release-pulsar-bookkeeper-0` now ?
2. Any events information?
3. Any logs?

**Comment by AliscaCL** [comment]
pod release-etcd-0/1/2 tells me that
0/3 nodes are available: 1 node had taint <node-role.kubernetes.io/master: > that the pod didn't tolerate, 2 node didn't find available persistent volumes to bind.

**Reply by AliscaCL** [reply]
I have build the pv for with local PV with node selector. 
And the local path has been already build. The disk space is also sufficient

**Reply by AliscaCL** [reply]
I have alter the parameter for memory, cpu, and storage, to make sure that the resource of the host is sufficient. 
It stille meet with this issue.

**Reply by jennyli-z** [reply]
> 

I haven't met that error before, but from the error message it is really because of the node has some issues that is not able to schedule the pod.
1. Try to create pvc by using the same storageClass as etcd's.
2. And then check `pv` is successfully created.

**Reply by jennyli-z** [reply]
Also, I have found one related issue essay https://stackoverflow.com/questions/63771336/cannot-get-pod-to-bind-local-storage-in-minikube-nodes-didnt-find-available , you can try to see if it can fix your problems
